synfig-studio/src/gui/canvasview.cpp

```diff
@@ -1431,12 +1431,11 @@ CanvasView::on_current_time_widget_changed()
 void
 CanvasView::on_set_end_time_widget_changed()
 {
-	get_canvas()->rend_desc().set_time_end(framedial->get_end_time());
-	//refresh the renddesc
-	refresh_rend_desc();
-	refresh_time_window();
-	//refresh canvas_properties
-	canvas_properties.refresh();
+	RendDesc rend_desc = get_canvas()->rend_desc();
+	rend_desc.set_time_end(framedial->get_end_time());
+
+	if(rend_desc.get_time_end() != get_canvas()->rend_desc().get_time_end())
```

Comment:
I think this `if` could be on the method beginning. You should then compare with framedial, as rend_desc wouldn't exist yet.